### DISCUSSION TITLE
How to install?

### DISCUSSION THREAD

**Discussion by warrenbreedlove** [Turbo87/sidebar-v2#294] [Q&A]
My coding and js skills are pretty weak. I was wondering if anyone could help by giving a little explanation of how to install this plugin into an existing leaflet project. Thanks! Im guessing its as simple as dragging and dropping some files in the right spots then linking in in the main html but nothing i try has worked thus far. Thanks!

**Comment by patrick-ryan** [comment]
I had the same question/experience. here's the steps I followed to get mine working. if I remember I'll update this to link to my project when I upload it.

setup:
- pure html/js running in my browser with basic index.html document
- deps: leaflet v1.8.0
- using [this example](https://github.com/Turbo87/sidebar-v2/blob/master/examples/position-right.html)

steps:
1. download [the js file](https://github.com/Turbo87/sidebar-v2/blob/master/js/leaflet-sidebar.js)
2. download [the css file](https://raw.githubusercontent.com/Turbo87/sidebar-v2/master/css/leaflet-sidebar.css)
3. update html page with elements copied from example html:
```html
<html>
    <head>
        <!-- leaflet css -->
        <!-- leaflet js -->
        
        <link href="https://maxcdn.bootstrapcdn.com/font-awesome/4.1.0/css/font-awesome.min.css" rel="stylesheet">

        <script src="./js/leaflet-sidebar.js"></script>
        <link rel="stylesheet" href="./css/leaflet-sidebar.css" />

        <style>
            #map { height: 100%; }
        </style>

    </head>
    <body>
        <div id="map"></div>

        <!-- copy #sidebar div from example html -->

    <script>
        // ...
        var map = L.map('map', {center: 0, zoom: 0, maxBounds: maxBounds}).fitBounds(maxBounds);

        // add layers or whatever

        L.control.sidebar('sidebar', {position: 'right'}).addTo(map);  // from example html
    </script>

    </body>

</html>
```

directory structure:
```shell
$ tree
.
├── css
│   └── leaflet-sidebar.css
├── index.html
├── js
│   └── leaflet-sidebar.js
```

seems to be working for me, needs some slight css adjustments but otherwise functional
